test/instances/Ring_finiteness.lean

```diff
@@ -0,0 +1,13 @@
+import Mathlib
+
+-- https://github.com/leanprover-community/mathlib4/pull/17557#issuecomment-2426920648
+
+variable (R : Type*) [Ring R] [IsSemisimpleRing R]
+
+/-- info: isNoetherian_of_isNoetherianRing_of_finite R R -/
+#guard_msgs in
+#synth IsNoetherianRing R
+
+/-- info: instIsArtinianOfIsSemisimpleModuleOfFinite -/
+#guard_msgs in
+#synth IsArtinianRing R
```

Comment:
Okay I've added the tests. The first result is not `IsSemisimpleModule.instIsNoetherianOfFinite` unlike the second result, probably because [that instance](https://github.com/leanprover-community/mathlib4/pull/17557/files#diff-95d7011cee766c6bd49b18c53feddf48cac70fc5b1a206eafa18d54b5bb6f5c0R226-R228) has low priority. I made the priority low because there's also an unconditional instance from `IsNoetherian` to `Module.Finite`.

Reply:
Oh! But I was asking for 
```lean
example : IsNoetherianRing := inferInstance 
example : IsArtinianRing := inferInstance 
```
The particular synthesis path is not as important as the fact that end users can synthesize the Artinian and Noetherian classes. 

Apologies for being unclear.

Reply:
I agree this is better! I was just copying from the other file and adjust.